ansible/roles/letsencrypt/templates/letsencrypt-http

```diff
@@ -1,12 +1,18 @@
-{% for domain in letsencrypt_domains %}
+
+# Generated by ansible
+# roles/letsencrypt/templates/letsencrypt-http
 
 server {
+    # Listen on port 80 for *any* domain
     listen 80;
-    server_name {{ domain }};
+    server_name _;
 
+    # Serve ACME challenge from disk
     location /.well-known/acme-challenge {
         root /var/www/letsencrypt;
         try_files $uri $uri/ =404;
     }
+
+    # Redirect everything else to port 443 regardless of domain
+    return 301 https://$host$request_uri;
```

Comment:
Yes this is a good idea. 

In case you haven't done so already, I suggest:
*  Checking if there are some services which require nginx to serve the service on port 80 and not enforce TLS (I don't think that is the case, but it's worth checking)
* Grepping for `c.location_letsencrypt()` and removing that template from there
* Dropping from `ansible/templates/common.j2` all functions which are no longer needed

Reply:
This PR is only for the letsencrypt role. Removing location_letsencrypt() in other roles should be done when we'll migrate the other roles to the new letsencrypt setup in future PRs.